packages/hash/frontend/src/blocks/onBlockLoaded.tsx

```diff
@@ -0,0 +1,77 @@
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import { getBlockDomId } from "./page/BlockView";
+
+type OnBlockLoadedFunction = (blockEntityId: string) => void;
+
+/** @private enforces use of custom provider */
+const BlockLoadedContext = createContext<OnBlockLoadedFunction | null>(null);
+
+export const BlockLoadedProvider: React.FC<{ routeHash: string }> = ({
+  routeHash,
+  children,
+}) => {
+  const [hashToScrollTo, setHashToScrollTo] = useState<string | null>(null);
+
+  const onBlockLoaded = useCallback((blockEntityId: string) => {
+    if (routeHash === getBlockDomId(blockEntityId)) {
+      setHashToScrollTo(routeHash);
```

Comment:
If a block reports that it has loaded in, and its `entityId` matches with the current `routeHash`, set it as the current block to be scrolled to.